### PR TITLE
Girder roles

### PR DESCRIPTION
**Pull request by opadron**
Eighth among a series of PR's.

Adds a new role: `girder`, for provisioning and managing girder nodes.

Initially contains the commits from all prior PR's in the series (will be rebased after they are merged).

New commits: 1

### CODE REVIEW

**Comment by opadron** [comment]
rebased after merging #21

File: roles/girder-install/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+      - libpng-dev
+      - libsqlite-dev
+      - libssl-dev
+      - libtool
+      - m4
+      - make
+      - mongodb
+      - ncurses-dev
+      - openjdk-7-jdk
+      - openjdk-7-jre-headless
```
Comment (zachmullen) [inline roles/girder-install/tasks/main.yml:33]: Why do we need gfortran, autoconf, sqlite, libexpat, mongodb, ncurses-dev, and openjdk?

Comment (opadron) [inline roles/girder-install/tasks/main.yml:33]: I don't think we do.  I just haven't been keeping close tabs on the dependencies.  Thanks for pointing it out.

**Comment by opadron** [comment]
I've added #28 to handle the issue of package dependencies in a more comprehensive manner.  Considering that the dependency issue spans multiple roles, many of which have already been merged from prior PRs, I propose we punt on the dependency issue and move the discussion to the new issue.

File: roles/girder-post-install/files/girder-post-install.py [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+    return method(*args, **kwds)
+
+def post(client, *args, **kwds):
+    return rest(client, 'post', *args, **kwds)
+
+def put(client, *args, **kwds):
+    return rest(client, 'put', *args, **kwds)
+
+def get(client, *args, **kwds):
+    return rest(client, 'get', *args, **kwds)
```
Comment (zachmullen) [inline roles/girder-post-install/files/girder-post-install.py]: We actually have all these methods inside `GirderClient` already. https://github.com/girder/girder/blob/master/clients/python/girder_client/__init__.py#L228-L238

**Comment by kotfic** [comment]
@opadron  @cpatrick I've been able to successfully deploy this role on a vagrant box and verify that everything is working. My concerns about this role fall into three categories:
1. inclusion of romanesco, climos_test and spark_test plugins which are not relevant to a general base installation of girder
2. Reliance on HDFS and HDFS asset stores to successfully run this role
3. Use of the `girder-post-install.py` file to configure the girder install

As I understand from @opadron  1 & 2 are needed for gobig and are known problems.  I would suggest adding refactoring issues about plugin and HDFS dependencies and then explicitly mention these in the girder documentation before the table of variables.  (E.g., _Note: This role depends on [plugins] and presence of a working HDFS installation -  please see issues [list issues]_).

My concern with 3 is the use of a non-trival python script to configure girder-  which should be the exclusive responsibility of the deployment mechanism (ansible). With that said,  reviewing covalic and HPCCloud deployments of girder+plugins it doesn't appear that there is a better solution than the ``girder-post-install` script.  I have started work on my own time to implement an ansible module wrapper around girder client that might resolve some of these issues.  In any case I don't think 3 should  block merging this branch.

If you can add the issues and a note in the documentation than this LGTM.

**Comment by opadron** [comment]
@kotfic we are totally in agreement.  I have added a new issue and updated the documentation for the girder role.  If you can make a final pass through these two items and are satisfied, I will pull the trigger on this PR and move on to our growing laundry list. :)

**Comment by kotfic** [comment]
LGTM

**Pull request merged**